src/salte-auth.js

```diff
@@ -1,1226 +1,323 @@
-        STATE_LOGIN: 'auth.state.login',
-        STATE_RENEW: 'auth.state.renew',
-        NONCE_IDTOKEN: 'auth.nonce.idtoken',
-        SESSION_STATE: 'auth.session.state',
-        USERNAME: 'auth.username',
-        IDTOKEN: 'auth.idtoken',
-        ERROR: 'auth.error',
-        ERROR_DESCRIPTION: 'auth.error.description',
-        LOGIN_REQUEST: 'auth.login.request',
-        LOGIN_ERROR: 'auth.login.error',
-        RENEW_STATUS: 'auth.token.renew.status'
-      },
-      RESOURCE_DELIMETER: '|',
-      LOADFRAME_TIMEOUT: '6000',
-      TOKEN_RENEW_STATUS_CANCELED: 'Canceled',
-      TOKEN_RENEW_STATUS_COMPLETED: 'Completed',
-      TOKEN_RENEW_STATUS_IN_PROGRESS: 'In Progress',
-      LOGGING_LEVEL: {
-        ERROR: 0,
-        WARN: 1,
-        INFO: 2,
-        VERBOSE: 3
-      },
-      LEVEL_STRING_MAP: {
-        0: 'ERROR:',
-        1: 'WARNING:',
-        2: 'INFO:',
-        3: 'VERBOSE:'
-      },
-      POPUP_WIDTH: 483,
-      POPUP_HEIGHT: 600
-    };
-
-    if (window.AuthenticationContext) {
-      return window.AuthenticationContext;
-    }
-    window.AuthenticationContext = this;
-
-    // public
-    this.config = {};
-    this.callback = null;
-    this.popUp = false;
-    this.isAngular = false;
-
-    // private
-    this._user = null;
-    this._activeRenewals = {};
-    this._loginInProgress = false;
-    this._renewStates = [];
-
-    this.callBackMappedToRenewStates = {};
-    this.callBacksMappedToRenewStates = {};
-
-    // validate before constructor assignments
-    if (config.displayCall && typeof config.displayCall !== 'function') {
-      throw new Error('displayCall is not a function');
-    }
-
-    if (!config.clientId) {
-      throw new Error('clientId is required');
-    }
-
-    if (!config.url || !config.url.match(/^https:\/\/.*\/$/)) {
-      throw new Error('url must be a valid https endpoint that ends in a forward slash.');
-    }
-
-    this.config = this._cloneConfig(config);
-
-    if (this.config.popUp) {
-      this.popUp = true;
-    }
-
-    if (this.config.callback && typeof this.config.callback === 'function') {
-      this.callback = this.config.callback;
-    }
-
-    // App can request idtoken for itself using clientid as resource
-    if (!this.config.loginResource) {
-      this.config.loginResource = this.config.clientId;
-    }
-
-    if (!this.config.redirectUri) {
-      this.config.redirectUri = window.location.href;
-    }
-
-    if (!this.config.anonymousEndpoints) {
-      this.config.anonymousEndpoints = [];
-    }
-
-    if (this.config.isAngular) {
-      this.isAngular = this.config.isAngular;
-    }
-
-    this.setResponseType(this.config.responseType);
-
-    if (this._isEmpty(this.config.hashPrefix)) {
-      this.config.hashPrefix = '';
-    }
+    if (window.salte.auth) {
+      return window.salte.auth;
+    }
+    window.salte.auth = this;
+
+    if (!config) {
+      throw new ReferenceError('A config must be provided.');
```

Comment:
All configuration errors should be identified before we set window.salte.auth to this.  Otherwise, subsequent calls may result in us returning a partially configured component.